### ISSUE TITLE
[BUG] Gather

### ISSUE THREAD

**Issue by sssimpleboy** [daquexian/onnx-simplifier#207]
**Describe the bug**
onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Non-zero status code returned while running Gather node. Name:'' Status Message: indices element out of data bounds, idx=0 must be within the inclusive range [0,-1]
**Model**
To reproduce the problem, please post download link of your model here, or send your model to daquexian566@gmail.com
After I converted my pth(96,544kb) to .onnx(98,446kb) , the error occured and after simplifying,the new onnx(97,607 kb) can't work as well and  the error still exsited.
![image](https://user-images.githubusercontent.com/100906198/180343738-7222457c-9fbf-4185-af4e-ffc6c5c78e11.png)

could you help me? thanks!

**Comment by sssimpleboy** [comment]
At first , grid_sample in torch seems can't  be converted to onnx, So I use belinear_grid_sample in mmcv to replace it.
I succeed creating a .onnx file ,but when I inference with the .onnx, the error occurs.

**Comment by sssimpleboy** [comment]
this is my aster_test_sim.onnx's result
![image](https://user-images.githubusercontent.com/100906198/180350967-360b78d5-c2f6-49db-844c-f9c7bfdb99ec.png)

**Comment by daquexian** [comment]
> After I converted my pth(96,544kb) to .onnx(98,446kb) , the error occured and after simplifying,the new onnx(97,607 kb) can't work as well and the error still exsited.

Sorry for the late response. Did you mean that the original onnx model (before simplifying) already has the error?

**Comment by sssimpleboy** [comment]
> > After I converted my pth(96,544kb) to .onnx(98,446kb) , the error occured and after simplifying,the new onnx(97,607 kb) can't work as well and the error still exsited.
> 
> Sorry for the late response. Did you mean that the original onnx model (before simplifying) already has the error?

yes

**Comment by sssimpleboy** [comment]
> > > After I converted my pth(96,544kb) to .onnx(98,446kb) , the error occured and after simplifying,the new onnx(97,607 kb) can't work as well and the error still exsited.
> > 
> > 
> > Sorry for the late response. Did you mean that the original onnx model (before simplifying) already has the error?
> 
> yes

I wanted to send my model to you.But considering that you are not familiar with this model,I don't if it is helpful

**Comment by sssimpleboy** [comment]
> > After I converted my pth(96,544kb) to .onnx(98,446kb) , the error occured and after simplifying,the new onnx(97,607 kb) can't work as well and the error still exsited.
> 
> Sorry for the late response. Did you mean that the original onnx model (before simplifying) already has the error?

last week I tried to devide the model to tps_part and encoder_decoder_part and create .onnx for them.I found that the tps_part onnx could work well , and the encoder_decoder_part can only work with the stable image which is used to create the onnx and for other images it will create the same  error about 'Gather' as before.
![image](https://user-images.githubusercontent.com/100906198/180674658-f3064f55-49f1-4a93-9325-f83f0fca745d.png)
I guess  that during converting,some variables were treated as constant,so the image changed the parameters couldn't change with the image and error occurs.
By the way, are you a Chinese,if you are ,we can chat with Chinese more conveniently? Your username loos like a Chinese name.

**Comment by daquexian** [comment]
> By the way, are you a Chinese？If you are ,we can chat with Chinese more conveniently? Your username loos like a Chinese name.

Yes 😂 Please join our QQ group 1021964010, the answer of verification question is "nndab"

**Comment by 1806610292** [comment]
> At first , grid_sample in torch seems can't be converted to onnx, So I use belinear_grid_sample in mmcv to replace it. I succeed creating a .onnx file ,but when I inference with the .onnx, the error occurs.

@sssimpleboy  hi, i have the same question, Have you solved it？

**Comment by sssimpleboy** [comment]
> > At first , grid_sample in torch seems can't be converted to onnx, So I use belinear_grid_sample in mmcv to replace it. I succeed creating a .onnx file ,but when I inference with the .onnx, the error occurs.
> 
> @sssimpleboy hi, i have the same question, Have you solved it？

no
